controller/Controller.php

```diff
@@ -517,4 +515,107 @@ function passwordPage($id)
         echo $view->render('views/password.html');
     }
 
+    /**
+     * Page to edit table items
+     *
+     * @param int $itemId The id of the item being edited
+     * @author Keller Flint
+     */
+    function editPage($itemId)
+    {
+        // Save Item
+        if (isset($_POST["itemSave"])) {
+            if ($this->_val->validateItem($_POST["itemName"])) {
+                $this->_db->updateItem($_POST["itemId"], $_POST["itemName"], $_POST["itemType"]);
+            } else {
+                $this->_f3->set("errors", $this->_val->getErrors());
+            }
+
+            if ($itemId == 0) {
+                if ($this->_val->validateItem($_POST["itemName"])) {
+                    $itemId = $this->_db->addItem($_POST["itemName"], $_POST["itemType"]);
+                    $this->_f3->reroute("edit/$itemId");
+                }
+            }
+        }
+
+        // Save State
+        if (isset($_POST["stateSave"])) {
+
+            // Changing isDone to an int
+            if (!isset($_POST["stateIsDone"])) {
+                $_POST["stateIsDone"] = 0;
+            } else {
+                $_POST["stateIsDone"] = 1;
+            }
+
+            // Update existing state
+            if ($_POST["stateId"] != 0) {
+                if ($this->_val->validateState($_POST["stateId"], $_POST["stateName"], $_POST["stateText"])) {
+                    $this->_db->updateState($_POST["stateId"], $_POST["stateName"], $_POST["stateSetBy"], $_POST["stateText"], $_POST["stateIsDone"]);
+                } else {
+                    $this->_f3->set("errors", $this->_val->getErrors());
+                }
+                // Add new state
+            } else {
+                if ($this->_val->validateState($_POST["stateId"], $_POST["stateName"], $_POST["stateText"])) {
+                    $this->_db->addState($itemId, $_POST["stateName"], $_POST["stateSetBy"], $_POST["stateText"], $_POST["stateIsDone"]);
+                } else {
+                    $this->_f3->set("errors", $this->_val->getErrors());
+                    $this->_f3->set("stateNameNew", $_POST["stateName"]);
+                    $this->_f3->set("stateSetByNew", $_POST["stateSetBy"]);
+                    $this->_f3->set("stateTextNew", $_POST["stateText"]);
+                    $this->_f3->set("stateIsDoneNew", $_POST["stateIsDone"]);
+                }
+            }
+
+        }
+
+        // Move Up
+        if (isset($_POST["moveUp"])) {
+            $this->_db->updateStateOrder($_POST["stateId"], -1);
```

Comment:
Could this be moved up too far? Should there be an if statement making sure it's not moved up more that it can be?

Reply:
The button to move up is not created unless there is an element above it